### PR TITLE
improve(snapshot):update testcase

### PR DESCRIPTION
**Pull request by ddaddaddda**

### CODE REVIEW

File: alicloud/data_source_alicloud_snapshots.go [hunk 1]
```diff
@@ -189,30 +187,31 @@ func dataSourceAlicloudSnapshotsRead(d *schema.ResourceData, meta interface{}) e
...
+	request.PageSize = requests.NewInteger(PageSizeLarge)
+	request.PageNumber = requests.NewInteger(1)
 	var allSnapshots []ecs.Snapshot
 	for {
 		raw, err := client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
-			return ecsClient.DescribeSnapshots(args)
+			return ecsClient.DescribeSnapshots(request)
 		})
 		if err != nil {
 			return WrapError(err)
```
Comment (YY-tomorrow) [inline alicloud/data_source_alicloud_snapshots.go]: please use WrapErrorf

File: alicloud/service_alicloud_ecs.go [hunk 2]
```diff
@@ -900,44 +900,49 @@ func (s *EcsService) QueryInstanceAllDisks(id string) ([]string, error) {
...
-	args := ecs.CreateDescribeSnapshotsRequest()
-	args.SnapshotIds = fmt.Sprintf("[\"%s\"]", snapshotId)
+func (s *EcsService) DescribeSnapshot(id string) (*ecs.Snapshot, error) {
+	request := ecs.CreateDescribeSnapshotsRequest()
+	request.SnapshotIds = fmt.Sprintf("[\"%s\"]", id)
 	raw, err := s.client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
-		return ecsClient.DescribeSnapshots(args)
+		return ecsClient.DescribeSnapshots(request)
 	})
 	if err != nil {
```
Comment (YY-tomorrow) [inline alicloud/service_alicloud_ecs.go]: check notfound?

File: alicloud/service_alicloud_ecs.go [hunk 3]
```diff
@@ -900,44 +900,49 @@ func (s *EcsService) QueryInstanceAllDisks(id string) ([]string, error) {
...
-		return nil, WrapErrorf(err, DefaultErrorMsg, snapshotId, args.GetActionName(), AlibabaCloudSdkGoERROR)
+		return nil, WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
 	}
```
Comment (YY-tomorrow) [inline alicloud/service_alicloud_ecs.go]: addDebug

File: alicloud/service_alicloud_ecs.go (hunk 2 above)
Comment (ddaddaddda) [reply alicloud/service_alicloud_ecs.go]: There is not exist  error code “notfound”  filter by snapshotId

File: alicloud/resource_alicloud_snapshot.go [hunk 4]
```diff
@@ -44,28 +49,32 @@ func resourceAliyunSnapshot() *schema.Resource {
...
-	resp := raw.(*ecs.CreateSnapshotResponse)
-	d.SetId(resp.SnapshotId)
+	addDebug(request.GetActionName(), raw)
+	response := raw.(*ecs.CreateSnapshotResponse)
+	d.SetId(response.SnapshotId)
 
 	ecsService := EcsService{client}
-	if err := ecsService.WaitForSnapshot(d.Id(), SnapshotCreatingAccomplished, DefaultLongTimeout); err != nil {
+
+	stateConf := BuildStateConf([]string{}, []string{string(SnapshotCreatingAccomplished)}, d.Timeout("create"), 0,
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_snapshot.go]: d.Timeout("create") ?

**Pull request merged**